pkg/jobs/hypershift/hypershift.go

```diff
@@ -0,0 +1,533 @@
+// Copyright Contributors to the Open Cluster Management project.
+package hypershift
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	clustercuratorv1 "github.com/stolostron/cluster-curator-controller/pkg/api/v1beta1"
+	"github.com/stolostron/cluster-curator-controller/pkg/jobs/utils"
+	managedclusterinfov1beta1 "github.com/stolostron/cluster-lifecycle-api/clusterinfo/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	clientv1 "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+/*
+We must use dynamic types here unfortunately because the Hypershift API requires
+an older version of sigs.k8s.io/controller-runtime/pkg/client(v0.13.1) which is
+not compatible with the version	that the Hive API requires and results in compile issues.
+*/
+func ActivateDeploy(dc dynamic.Interface, clusterName string, namespace string) error {
+	klog.V(0).Info("* Initiate Hypershift Provisioning")
+
+	// Update HostedCluster
+	err := removePausedUntil(dc, clusterName, namespace, utils.HCGVR)
+	if err != nil {
+		return err
+	}
+
+	// Update NodePool
+	// Need to account for 0 or multiple NodePools
+	nodePools, err := dc.Resource(utils.NPGVR).Namespace(namespace).List(context.TODO(), v1.ListOptions{})
+	utils.CheckError(err)
+
+	for _, np := range nodePools.Items {
```

Comment:
Can nodePools be nil here to cause a nilpointer? Maybe just add a nil check to be safe?

Reply:
@mikeshng I don't think nodePools can be nil here. I tried checking for nil but it won't let me, unless you have other ideas?
![image](https://github.com/stolostron/cluster-curator-controller/assets/38960034/f314e897-e3b1-4554-a9d1-fe5b8abf8055)

![image](https://github.com/stolostron/cluster-curator-controller/assets/38960034/909c68e8-6d02-4af1-ae54-f4ebf69bb1a2)

Reply:
Thanks for checking.